core/src/main/scala/org/broadinstitute/dsde/rawls/deltalayer/DeltaLayerTranslator.scala

```diff
@@ -0,0 +1,126 @@
+      case _ => None
+    }
+
+    // throw if we found anything other than AddUpdateAttribute in the operations
+    if (maybeAttributes.exists(_.isEmpty))
+      // in the future, may want to provide more info such as entity type/name and illegal operation
+      throw new DeltaLayerException(ERR_INVALID_OPERATION, code = BadRequest)
+
+    // flatten to remove the Options
+    val allValues: Seq[Attribute] = maybeAttributes.flatten
+
+    // validate nonEmpty/expected size of allValues
+    if (allValues.isEmpty)
+      // how did we reach here? This should be impossible
+      throw new DeltaLayerException(ERR_EMPTY_OPERATIONS, code = BadRequest)
+    if (allValues.size != allOps.size)
+      throw new DeltaLayerException("Unexpected error: count of update values did not match count of update operations",
+        code = InternalServerError)
+
+    // validate that all upsert values are string/boolean/number only
+    val (supportedTypes, unsupportedTypes) = allValues partition isSupportedDataType
+
+    if (unsupportedTypes.nonEmpty)
+      throw new DeltaLayerException(ERR_INVALID_DATATYPE, code = BadRequest)
+
+    if (supportedTypes.isEmpty)
+      // how did we reach here? This should be impossible
+      throw new DeltaLayerException(ERR_INVALID_DATATYPE, code = BadRequest)
+
+    // everything validated; return the original update definitions
+    entityUpdates
+  }
+
+  /**
+   * is the supplied Attribute supported by Delta Layer?
+   * @param attr the Attribute to inspect
+   * @return whether or not Delta Layer supports this Attribute's type
+   */
+  private def isSupportedDataType(attr: Attribute): Boolean = attr match {
+    case AttributeBoolean(_) | AttributeNumber(_) | AttributeString(_) | AttributeNull | AttributeValueEmptyList => true
+    case AttributeValueList(elems) => elems forall isSupportedDataType
+    case _ =>
+      // AttributeNull, AttributeValueRawJson, AttributeEntityReference,
+      // AttributeEntityReferenceList, AttributeEntityReferenceEmptyList
+      false
+  }
+
+
+  /**
+   * Transforms the Rawls user-visible model classes into the models we will write
+   * into Delta Layer insert files. Calls validateEntityUpdates before performing
+   * transformations
+   *
+   * @param entityUpdates the updates to translate
+   * @return the collection of DeltaRow objects to write to a Delta Layer insert file
+   */
+  def translateEntityUpdates(entityUpdates: Seq[EntityUpdateDefinition]): Seq[DeltaRow] = {
+
+    val validUpdates = validateEntityUpdates(entityUpdates)
+
+    validUpdates.flatMap { update =>
+      // we expect all calls to use datarepo_row_id as entity name
+      val datarepoRowId = Try(UUID.fromString(update.name)) match {
+        case Success(id) => id
+        case Failure(_) => throw new DeltaLayerException(ERR_INVALID_ENTITYNAME, code = BadRequest)
+      }
+      update.operations map { op =>
+        // strip "default:" from attr name
+        val name = toDelimitedName(op.name)
```

Comment:
Curious as to why we remove the default namespace qualifier?

Reply:
we remove the default namespace qualifier mostly for consistency with elsewhere in the stack. If you view the Data tab of a workspace, for instance, all the columns you see in a data table are "foo" and "bar," not "default:foo" and "default:bar."